src/nodes/branch.rs

```diff
@@ -193,6 +201,50 @@ impl<'a> BranchNodeRef<'a> {
     }
 }
 
+/// Iterator over branch node children.
+#[derive(Debug)]
+pub struct BranchChildrenIter<'a> {
+    pos: u8,
+    end: u8,
+    stack_ptr: usize,
+    state_mask: &'a TrieMask,
+    stack: &'a [Vec<u8>],
+}
+
+impl<'a> BranchChildrenIter<'a> {
+    /// Create new iterator over branch node children.
+    pub fn new(node: &BranchNodeRef<'a>) -> Self {
```

Comment:
should probably be private and a method on BranchNodeRef?

Reply:
no harm in having it public

Reply:
right, but you would want `node.children()` rather than `BranchChildrenIter::new(node)`

Reply:
that's already exposed via `indexed_children`, but alright, point taken, will hide vis